src/gui/qgsmaptoolcapture.cpp

```diff
@@ -585,6 +586,12 @@ void QgsMapToolCapture::undo()
     mCaptureCurve.deleteVertex( vertexToRemove );
     mSnappingMatches.removeAt( vertexToRemove.vertex );
 
+    // If Cad is enabled, update the points accordingly
+    if ( mCadDockWidget->cadEnabled() )
```

Comment:
Is this check necessary?
Also I don't recall what happens if you disable and reenable the dock, are your points kept? If so, we should not check then.

Reply:
I think you're right, will remove the check